File: app/src/main/java/com/ephirium/storyline/feature/model/Post.java
package com.ephirium.storyline.feature.model;

public class Post {

    private final String fileSource;
    private final String name;
    private final String description;

    public Post(String name, String description, String fileSource){
        this.name = name;
        this.description = description;
        this.fileSource = fileSource;
    }

    public String getName() {
        return name;
    }
    public String getDescription() {
        return description;
    }
    public String getFileSource(){
        return fileSource;
    }
}

File: app/src/main/java/com/ephirium/storyline/model/Post.kt
package com.ephirium.storyline.model

import android.graphics.drawable.Drawable
import com.ephirium.common.delegate.DelegateItem

data class Post(val id: String, val name: String, val description: String, val fileSource: String) :
    DelegateItem {
    var drawable: Drawable? = null
    override fun getDelegateId(): String {
        return id
    }

    override fun getDelegateContent(): String {
        return name + description + fileSource
    }
}
